src/mono/mono/mini/mini-trampolines.c

```diff
@@ -1063,18 +1063,11 @@ mono_delegate_trampoline (host_mgreg_t *regs, guint8 *code, gpointer *arg, guint
 			}
 		}
 
-		if (delegate->method_ptr == NULL && tramp_info->method == NULL && delegate->target != NULL && method->flags & METHOD_ATTRIBUTE_VIRTUAL) {
-			/* tramp_info->method == NULL happens when someone asks us to JIT some delegate's
-			 * Invoke method (see compile_special).  In that case if method is virtual, the target
-			 * could be some derived class, so we need to find the correct override.
+		if (delegate->method_is_virtual) {
```

Comment:
Shouldn't we still do the check if `delegate->target` is non null, otherwise we crash during lookup ?

Reply:
There is a null check emitted by handle_delegate_ctor ().

Reply:
Isn't that just the fastpath delegate creation ? So we might have delegates from other sources with NULL targets.

Reply:
method_is_virtual is only set by the fast path which does a null check.